### PR TITLE
Remove external_redis.port from migration script since not used since v1.10.0

### PR DESCRIPTION
**Pull request by sixeIa**
Hello,

Trying to update Harbor from v2.1.2 to 2.2.0, we faced an issue due to how jinja2 template is made.
We are using redis+sentinel and we were unable to make the migration works.

Navigating to the jinja2 template you made, it allowed me to see a field "{{ external_redis.port }}" was used but removed since v1.10.0 (version since when you're supporting redis and redis+sentinel).

Anyway, removing this field allowed me to migrate my harbor.yml file.

I don't know if there are side-effects for such a modification but don't hesitate to tell me if there is something more to change.

Alexis

_(I made a PR few minutes ago but she wasn't following your contribution requirements)_

### CODE REVIEW

**Comment by sixeIa** [comment]
Hello,

Any news about this PR ? I saw on Slack that other people where facing the same issue.

Thank you,

**Comment by bitsf** [comment]
this `external_redis.port` is just used for migrate from 2.0.x to 2.1.x, so it is not correct after 2.1.
however if user upgrade from 2.0.x to 2.2.x directly, the fix would still have problem.

@ninjadq can you check how should this be fixed in migrate script

**Comment by sixeIa** [comment]
Hi,

Oh yep okay, so there is an issue when bumping version as you told.
If you have any idea on how I can improve it, don't hesitate to tell me.

File: make/photon/prepare/migrations/version_2_0_0/harbor.yml.jinja [hunk 1]
```diff
@@ -323,7 +323,7 @@ external_redis:
   # host for redis: <host_redis>:<port_redis>
   # host for redis+sentinel:
   #  <host_sentinel1>:<port_sentinel1>,<host_sentinel2>:<port_sentinel2>,<host_sentinel3>:<port_sentinel3>
-  host: {{ external_redis.host }}:{{ external_redis.port }}
+  host: {{ external_redis.host }}
```
Comment (ninjadq) [inline make/photon/prepare/migrations/version_2_0_0/harbor.yml.jinja:326]: it didn't need to remove `{{ external_redis.port }}`

File: make/photon/prepare/migrations/version_2_1_0/harbor.yml.jinja [hunk 2]
```diff
@@ -373,7 +373,7 @@ external_redis:
   # host for redis: <host_redis>:<port_redis>
   # host for redis+sentinel:
   #  <host_sentinel1>:<port_sentinel1>,<host_sentinel2>:<port_sentinel2>,<host_sentinel3>:<port_sentinel3>
-  host: {{ external_redis.host }}:{{ external_redis.port }}
+  host: {{ external_redis.host }}
```
Comment (ninjadq) [inline make/photon/prepare/migrations/version_2_1_0/harbor.yml.jinja:376]: the same as above, `external_redis.port` must exist here.

File: make/photon/prepare/migrations/version_2_2_0/harbor.yml.jinja [hunk 3]
```diff
@@ -350,7 +350,7 @@ external_redis:
   # host for redis: <host_redis>:<port_redis>
   # host for redis+sentinel:
   #  <host_sentinel1>:<port_sentinel1>,<host_sentinel2>:<port_sentinel2>,<host_sentinel3>:<port_sentinel3>
-  host: {{ external_redis.host }}:{{ external_redis.port }}
+  host: {{ external_redis.host }}
```
Comment (ninjadq) [inline make/photon/prepare/migrations/version_2_2_0/harbor.yml.jinja:353]: only remove `:{{ external_redis.port }}` here is enough

**Review by ninjadq — CHANGES_REQUESTED** [review]

**Review by sixeIa — COMMENTED** [review]
Modifications done

**Review by ninjadq — APPROVED** [review]
lgtm

**Review by bitsf — APPROVED** [review]

**Comment by wilsonge** [comment]
@bitsf Thanks for merging this. Is there any chance this can be cherry-picked back into the 2.2 branch as otherwise we won't be able to take advantage of this fix until 2.3 obviously...

**Pull request merged**